django/db/models/query.py

```diff
@@ -473,6 +473,47 @@ def bulk_create(self, objs, batch_size=None, ignore_conflicts=False):
 
         return objs
 
+    def bulk_save(self, objs, fields, batch_size=None):
+        """
+        Update the given fields in each of the given objects in the database.
+        (Doesn't call save() on each of the instances.)
+        """
+        if batch_size is not None and batch_size < 0:
+            raise ValueError('Batch size must be a positive integer.')
+        if not fields:
+            raise ValueError('Specific field names must be given to bulk_save().')
+        objs = tuple(objs)
+        if not all(obj.pk for obj in objs):
+            raise ValueError('All objects must have a primary key set.')
+        if not objs:
+            return
+        fields = [self.model._meta.get_field(name) for name in fields]
+        if any(not f.concrete or (f.is_relation and f.many_to_many) for f in fields):
+            raise ValueError('bulk_save() can only be used with concrete fields.')
+        ops = connections[self.db].ops
+        # PK is used twice in the resulting update query, once in the filter
+        # and once in the WHEN. Each field will also have one CAST.
+        max_batch_size = ops.bulk_batch_size(['pk', 'pk'] + fields, objs)
```

Comment:
I might be wrong (currently struggling to wrap my head around bulk_bath_size), but isn't it one WHEN clause and therefore a variable for ``pk`` *per field*, so this should read more like ``['pk'] + 2 * fields``?

Reply:
@raphaelm This line is correct. We use `pk` twice for each object, once in a `WHEN id = ... THEN` clause and once in a `WHERE id IN (...)` clause to filter rows for update.